packages/flame/lib/src/components/custom_painter_component.dart

```diff
@@ -0,0 +1,33 @@
+import 'package:flutter/widgets.dart';
+
+import '../extensions/vector2.dart';
+import 'position_component.dart';
+
+/// A [PositionComponent] that renders a [CustomPainter] at the designated
+/// position, scaled to have the designated size and rotated to the specified
+/// angle.
+///
+/// This component is usefull to [CustomPainter]s between your Flutter UI and Flame game.
+class CustomPainterComponent extends PositionComponent {
+  /// The [CustomPainter] used to render this component
+  CustomPainter? painter;
+
+  CustomPainterComponent({
+    Vector2? position,
+    Vector2? size,
+    int? priority,
+    this.painter,
```

Comment:
Should we not have this as `@required` and the global field as non-nullable?

Reply:
I would have this as the first argument too.

Reply:
I am following the way we do on sprite and sprite animation component.

Keeping this as optional we allow people to both pass the painter on the constructor, or when extending this class, init it on the `onLoad` method. Which is what we have been doing latelly.